serverless_cluster.go

```diff
@@ -0,0 +1,156 @@
+package hazelcastcloud
+
+import (
+	"context"
+	"github.com/hazelcast/hazelcast-cloud-sdk-go/models"
+)
+
+// ServerlessClusterService is used to interact with serverless clusters.
+type ServerlessClusterService struct {
+	client *Client
+}
+
+// NewServerlessClusterService returns a new instance of the service.
+func NewServerlessClusterService(client *Client) ServerlessClusterService {
+	return ServerlessClusterService{client}
+}
+
+// Create creates a serverless cluster according to configuration provided in the request.
+func (svc ServerlessClusterService) Create(ctx context.Context, input *models.CreateServerlessClusterInput) (*models.Cluster, *Response, error) {
+	var cluster models.Cluster
+	var graphqlRequest = models.GraphqlRequest{
+		Operation: models.Mutation,
+		Name:      "createServerlessCluster",
+		Input:     *input,
+		Args:      nil,
+		Response:  cluster,
+	}
+	req, err := svc.client.NewRequest(&graphqlRequest)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	resp, err := svc.client.Do(ctx, req, &cluster)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	return &cluster, resp, nil
+}
+
+// List provides the ability to get a list of serverless clusters.
+func (svc ServerlessClusterService) List(ctx context.Context) (*[]models.Cluster, *Response, error) {
+	var clusterList []models.Cluster
+	graphqlRequest := models.GraphqlRequest{
+		Name:      "clusters",
+		Operation: models.Query,
+		Input:     nil,
+		Args: models.ClusterListInput{
+			ProductType: models.Starter,
+		},
+		Response: clusterList,
+	}
+	req, err := svc.client.NewRequest(&graphqlRequest)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	resp, err := svc.client.Do(ctx, req, &clusterList)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	return &clusterList, resp, err
```

Comment:
does it make sense to create a separate task to do postfiltering by plan_name? and also to make request in graphql unpaged.

Reply:
I think this is the question to @huseyinbabal . As long we are going to disable `standard` clusters, this method should return both `standard` and `serverless` clusters. But in fact will return just `serverless`. But anyway I'd like to know what @huseyinbabal thinks about that.